structured_data/2021_06_07_timeseries_labeling_tools/README.md

```diff
@@ -0,0 +1,282 @@
+Within these projects different *label tasks* can be created. 
+Label tasks are time series segments that need to be labeled.
+Once the operator is happy with the labelling, it can be submitted.
+
+#### Import
+To import data, a data template template needs to be made. 
+This template is basically the configuration of how the data looks like (which columns are important), but also how the data can be viewed during labelling.
+This allows for a very flexible import system.
+More info on these data templates: [https://labelstud.io/templates/time_series.html](https://labelstud.io/templates/time_series.html)
+
+The data needs to be formatted in wide format, which for certain cases might lead to sparse datasets.
+On top of that, Label Studio seems to set missing values (~ sparse data) to 0, complicating importing data from a single source, for data that was not taken at the same time.
+To avoid this, multiple different files can be used.
+
+The tool can use different data sources. For an overview check: [https://labelstud.io/guide/storedata.html](https://labelstud.io/guide/storedata.html).
+As an example, Label Studio can directly connect to a GCP (storage bucket): [https://labelstud.io/guide/storage.html#Google-Cloud-Storage](https://labelstud.io/guide/storage.html#Google-Cloud-Storage).
+Unfortunately this did not seem to work outside of the VPC. This is probably due to a rights issue.
+
+Data can be imported using the GUI or using an API.
+Files up to 200Mb are supposed to work, although it started failing above 50Mb using the GUI.
+Files that are bigger suffer from a *validation error*.
+
+Label Studio has an API that can be used to pump data in.  ([https://labelstud.io/guide/tasks.html#Import-data-using-the-API](https://labelstud.io/guide/tasks.html#Import-data-using-the-API))
+
+Some other tips:
+- Data should be sorted on time
+- Missing data: data will be set to 0 (flat line).
+- Corrupt data (e.g. strings): time series just stop at the first time point it encountered a string. So even if the data afterwards is okay. There is simply no data to been seen (empty time series plot).
+
+So be a bit careful. A small anomaly might destroy the rest of the time series. Also the sparse data part is something to keep an eye on.
+
+#### Export
+Data can be exported in different ways, JSON making most sense. There is also an API, which allows to automate the export process + parsing + dumping towards e.g. GCP.
+
+[https://labelstud.io/templates/time_series.html#Output-format-example](https://labelstud.io/templates/time_series.html#Output-format-example)
+
+#### Playground
+Online sandbox to play around.
+[https://labelstud.io/playground/](https://labelstud.io/playground/)
+
+#### Conclusion
+In general the whole interface, with users and tasks make it really interesting to work with and to organize it when having multiple users.
+The tool also has a some flexibility regarding backend data storage as well as for importing data from different sources.
```

Comment:
Didn't you mention this before already? About Postgres/SQLite?

Reply:
yes, I feel it is still valid in the conclusion?